specification/confidentialledger/data-plane/Microsoft.ConfidentialLedger/stable/2022-05-13/confidentialledger.json

```
@@ -0,0 +1,949 @@
+          }
+        },
+        "summary": "Gets the consortium members.",
+        "description": "Consortium members can manage the Confidential Ledger.",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of consortium members.",
+            "schema": {
+              "$ref": "#/definitions/Consortium"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      }
+    },
+    "/app/enclaveQuotes": {
+      "get": {
+        "operationId": "ConfidentialLedger_GetEnclaveQuotes",
+        "x-ms-examples": {
+          "GetEnclaveQuotes": {
+            "$ref": "./examples/ledger/GetEnclaveQuotes.json"
+          }
+        },
+        "summary": "Gets quotes for all nodes of the Confidential Ledger.",
+        "description": "A quote is an SGX enclave measurement that can be used to verify the validity of a node and its enclave.",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Response containing enclave quotes.",
+            "schema": {
+              "$ref": "#/definitions/ConfidentialLedgerEnclaves"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      }
+    },
+    "/app/collections": {
+      "get": {
+        "operationId": "ConfidentialLedger_GetCollections",
+        "x-ms-examples": {
+          "GetCollectionIds": {
+            "$ref": "./examples/ledger/GetCollections.json"
+          }
+        },
+        "summary": "Retrieves a list of collection ids present in the Confidential Ledger",
+        "description": "Collection ids are user-created collections of ledger entries",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of collections",
+            "schema": {
+              "$ref": "#/definitions/CollectionArray"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      }
+    },
+    "/app/transactions": {
+      "get": {
+        "operationId": "ConfidentialLedger_GetLedgerEntries",
```

The general convention for the operationId of a get method on a collection is to use the verb "List" rather than "Get" -- this helps differentiate it from the "Get" method on a single element of the collection.
```suggestion
        "operationId": "ConfidentialLedger_ListLedgerEntries",
```

---

Should I make the same change for `GetCollections`?

---

Yes please.  And ... I'm sorry I missed this earlier ... but we really discourage returning a bare array in the response body. This should really be structured as an object with a `value` property that contains the array.  I know this is very late ... is there any possibility of fixing that?  I will completely understand if you say no ...

---

No worries at all! I will look into how viable it is to fix that.

---

Hi, @mikekistler, unfortunately we are in code freeze for our TPAL app. Since we would need to change the TPAL app to implement the response structure you described, this change could destabilize our service from the ground up. I will, however, add this change to our backlog so that we can work on it after GA.

---

I don't think you can make this change after you GA.  It will be a "breaking change" which is an onerous process.  Since the point of making this an object was to _avoid_ a potential breaking change in the future, it is best to leave it an array at this point. Sorry I didn't catch this earlier but now I know to be more diligent about this going forward.